dsn.go

```diff
@@ -98,6 +100,13 @@ func DSN(cfg *Config) (dsn string, err error) {
 	if cfg.Application != clientType {
 		params.Add("application", cfg.Application)
 	}
+	if cfg.Token != "" {
+		escapedToken, err := url.QueryUnescape(cfg.Token)
```

Comment:
https://github.com/snowflakedb/gosnowflake/blob/0285e80731d13676d55c1929f8b7d71e42d2d897/dsn.go#L373 already unescaped the value before cfg.Token is assigned. Is there any reason why unescape here again?

Reply:
Other than the above comment. Looks good to me.

Reply:
Ah I didn't see that earlier QueryUnescape... just removed mine and re-ran the tests and it works! Will update the branch accordingly.